relay/p2p/broadcaster.go

```diff
@@ -191,19 +219,38 @@ func (b *broadcaster) broadcastCurrentSignatures(peerId core.PeerID) error {
 	return nil
 }
 
-func (b *broadcaster) sendSignedMessageToPeer(msg *SignedMessage, peerId core.PeerID) error {
+func (b *broadcaster) retrieveUniqueMessages() map[string]*core.SignedMessage {
+	allMessages := make(map[string]*core.SignedMessage)
+	for _, client := range b.clients {
+		messages := client.AllStoredSignatures()
+		b.processMessages(allMessages, messages)
+	}
+
+	return allMessages
+}
+
+func (b *broadcaster) processMessages(
+	allMessages map[string]*core.SignedMessage,
+	clientMessages []*core.SignedMessage,
+) {
+	for _, msg := range clientMessages {
```

Comment:
why this function doesn't return a new map instead of modifying the one received in the parameters? Also, on `retrieveUniqueMessages()` you create a new map and send it here. If the map was already populated it would have had sense, but in this use case, I think you should refactor

Reply:
was split as to not have 2 nested fors. Changed so I won't pass a map as an input parameter that I will change inside the function. Hope it is clear.